### PR TITLE
Some minor nits

### PR DESCRIPTION
**Pull request by nyurik**
Note that I am not certain if `write_all(str)` is faster than the `write!` macro. I know for a fact that it is faster in case of the `format!("{}", value)` vs `v.write_str(value)`

### CODE REVIEW

**Review by jayvdb — APPROVED** [review]

File: src/bin/run-parser-test-suite.rs [hunk 1]
```diff
@@ -142,7 +143,7 @@ fn print_escaped(stdout: &mut dyn Write, s: &str) {
 fn main() -> ExitCode {
     let args = env::args_os().skip(1);
     if args.len() == 0 {
-        let _ = writeln!(io::stderr(), "Usage: run-parser-test-suite <in.yaml>...");
+        _ = io::stderr().write_all("Usage: run-parser-test-suite <in.yaml>...".as_bytes());
```
Comment (jayvdb) [inline src/bin/run-parser-test-suite.rs]: I think this is a case where [`eprintln`](https://doc.rust-lang.org/std/macro.eprintln.html#) would be better for simplicity - any perf improvement isnt worth it in this context.

**Comment by jayvdb** [comment]
re perf , I ran the bench on master, and then on this branch, and the results dont look good.

```text
> cargo bench
...
     Running benches/bench.rs (target/release/deps/bench-30519995111badcc)
libyaml-safer parse large
                        time:   [21.530 ms 21.651 ms 21.785 ms]
                        change: [+13.243% +13.914% +14.748%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 4 outliers among 100 measurements (4.00%)
  2 (2.00%) low severe
  1 (1.00%) high mild
  1 (1.00%) high severe

unsafe-libyaml parse large
                        time:   [22.219 ms 22.410 ms 22.581 ms]
                        change: [+2.9812% +4.0982% +5.0871%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 17 outliers among 100 measurements (17.00%)
  10 (10.00%) low severe
  4 (4.00%) high mild
  3 (3.00%) high severe

libyaml-safer emit large
                        time:   [13.020 ms 13.192 ms 13.373 ms]
                        change: [+0.8594% +2.2021% +3.7700%] (p = 0.00 < 0.05)
                        Change within noise threshold.

unsafe-libyaml emit large
                        time:   [12.839 ms 13.064 ms 13.294 ms]
                        change: [+1.4694% +3.7927% +6.1532%] (p = 0.00 < 0.05)
                        Performance has regressed.
```

How much of that is a result of using `write_all(str)` .. ?

Or the introduction of variable `prefix` ..?

File: src/bin/run-parser-test-suite.rs (hunk 1 above)
Comment (nyurik) [reply src/bin/run-parser-test-suite.rs]: good catch, makes sense, i fixed it in a few places

**Comment by nyurik** [comment]
Interesting...  so the `write_all` are all inside the `src/bin/run-parser-test-suite.rs` -- so they shouldn't actually be part of testing - come to think of it, we may want to keep them as is because they shouldn't be in production anyway, and I am not certain why there is all this extra code for dealing with `stdout/stderr` - shouldn't it just use `println/eprintln`?

The prefix is the only thing that is actually in the code path... and if the compiler is unable to optimize that, seems like a more substantial problem... Can you try to inline suffix and prefix inside the `format!` call, and see if you get a different perf result?

**Comment by jayvdb** [comment]
Hmm.. I've re-run `cargo bench` a few times and the perf regresses without any code changes .

**Comment by simonask** [comment]
Overall I think these changes are fine, except that I don't really think there's a compelling reason to replace `writeln!()` with the arguably less idiomatic `write_all`, only because this is just part of the test runner, so microoptimizing it doesn't make a lot of sense.

> Hmm.. I've re-run `cargo bench` a few times and the perf regresses without any code changes .

Hm, I think the benchmark may be quite sensitive to scheduling due to its size. In its current state, it may only really be suitable for comparing the overall performance to `unsafe-libyaml`.

**Comment by nyurik** [comment]
agree, i will revert the writeln - much more sensible here

**Pull request closed without merging**